include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -249,6 +263,44 @@ class fm_index
         // index.m_sigma = largest_char;
     }
 
+    //!\overload
+    void construct(text_t const & text)
+        //!\cond
+        requires is_collection
+        //!\endcond
+    {
+        // text must not be empty
+        if (text.begin() == text.end())
+            throw std::invalid_argument("The text that is indexed cannot be empty.");
+        this->text = &text;
+
+        concatenated_sequences<value_type_t<text_t>> cvec{text};
```

Comment:
Check whether cvec is empty too? The above check only check whether the vector of vectors is non-empty but not if the contained vectors are also non-empty.